AsyncDisplayKit/Layout/ASLayout.mm

```diff
@@ -88,6 +94,14 @@ - (instancetype)initWithLayoutElement:(id<ASLayoutElement>)layoutElement
 
     _sublayouts = sublayouts != nil ? [sublayouts copy] : @[];
     _flattened = NO;
+    
+    // Add sublayouts layout elements to an internal array to retain it while the layout lives
+    if (sublayouts.count > 0) {
```

Comment:
@maicki this is pretty awesome!  Would it be possible to put this retain operation into the setIsFlattened:YES?  Then we could assert that all sublayouts are nodes, and it would avoid any array and retain overhead for the many ASLayoutSpecs that are created during the measurement process.

The code that would trigger it is below, but I think the retain operation would happen in the flattened setter.  I think this would still make sense from a logical / usage perspective, certainly because this is a) an internal behavior and b) all externally accessible long-lived ASLayouts are flattened.

       ASLayout *layout = [ASLayout layoutWithLayout:context.layout position:context.absolutePosition];
       layout.flattened = YES;

The one question is: what happens if a user calls layoutThatFits: and saves a reference to the resulting ASLayout?  Do we always flatten those layouts?  If so, we're good.